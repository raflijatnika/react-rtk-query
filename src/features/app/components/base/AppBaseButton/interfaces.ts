// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
