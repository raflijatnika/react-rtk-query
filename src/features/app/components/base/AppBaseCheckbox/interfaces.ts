// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
