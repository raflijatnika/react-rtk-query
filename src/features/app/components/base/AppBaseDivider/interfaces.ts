// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
