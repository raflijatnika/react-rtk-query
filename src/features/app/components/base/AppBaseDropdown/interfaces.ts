// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
