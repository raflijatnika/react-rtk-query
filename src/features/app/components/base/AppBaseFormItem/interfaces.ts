// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
