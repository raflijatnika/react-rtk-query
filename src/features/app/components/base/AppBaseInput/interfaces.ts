// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
