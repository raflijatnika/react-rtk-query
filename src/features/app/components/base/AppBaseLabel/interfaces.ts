// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
