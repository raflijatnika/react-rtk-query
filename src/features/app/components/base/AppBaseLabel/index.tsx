// Styled Components
import styled from 'styled-components'

// Constant
import { APP_COLOR } from '@/features/app/constant/app-style.constant'

// Interfaces
import { IAppBaseLabelProps } from './interfaces'

const AppBaseLabel = styled.label<IAppBaseLabelProps>`
  font-size: ${props =>
    props?.fontSize ? `${props.fontSize}px` : '11px'} !important;
  font-weight: ${props =>
    props?.fontWeight ? props.fontWeight : props?.isBold ? 700 : 400};
  color: ${props => (props?.fontColor ? props.fontColor : APP_COLOR.DARK)};
`

export { AppBaseLabel }
