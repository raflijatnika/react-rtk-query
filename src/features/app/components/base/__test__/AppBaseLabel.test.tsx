// Test
import { render } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseLabel } from '../AppBaseLabel'

// Interfaces
import { IAppBaseLabelProps } from '../AppBaseLabel/interfaces'

// Constant
import { APP_COLOR } from '@/features/app/constant/app-style.constant'

const rendererInstance = (props?: IAppBaseLabelProps): ReactTestRenderer =>
  renderer.create(<AppBaseLabel {...props} />)

it('AppBaseLabel should render', () => {
  render(<AppBaseLabel />)
})

it('AppBaseLabel should have snapshot', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toMatchSnapshot()
})

it('AppBaseLabel should have default style', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toHaveStyleRule('font-size', '11px !important')
  expect(tree).toHaveStyleRule('font-weight', '400')
  expect(tree).toHaveStyleRule('color', APP_COLOR.DARK)
})

it('AppBaseLabel should correct props', () => {
  const tree = rendererInstance({
    fontSize: 16,
    fontWeight: 700,
    fontColor: APP_COLOR.WHITE
  }).toJSON()
  const secondTree = rendererInstance({ isBold: true }).toJSON()

  expect(tree).toHaveStyleRule('font-size', '16px !important')
  expect(tree).toHaveStyleRule('font-weight', '700')
  expect(tree).toHaveStyleRule('color', APP_COLOR.WHITE)
  expect(secondTree).toHaveStyleRule('font-weight', '700')
})
