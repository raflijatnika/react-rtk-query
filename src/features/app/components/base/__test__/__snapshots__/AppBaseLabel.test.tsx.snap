// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseLabel should have snapshot 1`] = `
.c0 {
  font-size: 11px !important;
  font-weight: 400;
  color: #06152B;
}

<label
  className="c0"
/>
`;
